### PR TITLE
Display help text on empty subcommand by default

### PR DESCRIPTION
**Pull request by markusthoemmes**
#### What this PR does / why we need it:

This is a small quality-of-life improvement, where instead of guiding the user to use `--help` when they didn't enter a specific subcommand, it shows the help text by default.

https://github.com/kubernetes/kubernetes/blob/efce40b931b5250498b9d8bcd67f987095ab6f74/hack/lib/util.sh#L714 in Kubernetes inspired me to do this. The `docker` CLI is behaving this way as well and Kubernetes uses this to check if `docker buildx` exists. I suppose the K8s side could potentially change in this case as well, but I wanted to see if y'all think this is a worthwhile improvement for human users anyway.

#### How to verify it

Run `podman`. It used to output

```
Error: missing command 'podman COMMAND'
Try 'podman --help' for more information.
```

With this it prints the help text immediately.

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

I realize that this might be contentious in nature and I haven't found an issue relating to this. Please let me know if we should instead open an issue and discuss there first.

### CODE REVIEW

**Comment by rhatdan** [comment]
We should probably do this on subcommands as well, what I don't like is scroll off the screen when you specify a bad option.
```
 $ podman network
Error: missing command 'podman network COMMAND'
Try 'podman network --help' for more information.
$ podman container
Error: missing command 'podman container COMMAND'
Try 'podman container --help' for more information.
```

Git does a nice job at this.

Docker does a better job then us on this also.

Both Git and Docker stack the options at the top and the commands at the bottom, which is probably something we should consider.

**Comment by rhatdan** [comment]
If it does that would be good.

**Review by jwhonce — CHANGES_REQUESTED** [review]
Thanks for the PR. Just one change to restore the original semantics.

File: cmd/podman/validate/args.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ func SubCommandExists(cmd *cobra.Command, args []string) error {
 		}
 		return errors.Errorf("unrecognized command `%[1]s %[2]s`\n\nDid you mean this?\n\t%[3]s\n\nTry '%[1]s --help' for more information.", cmd.CommandPath(), args[0], strings.Join(suggestions, "\n\t"))
 	}
-	return errors.Errorf("missing command '%[1]s COMMAND'\nTry '%[1]s --help' for more information.", cmd.CommandPath())
+	return cmd.Help()
```
[the remark this answers is not in the document]
Comment (rhatdan) [reply cmd/podman/validate/args.go:30]: Good catch @jwhonce

**Comment by markusthoemmes** [comment]
@rhatdan @jwhonce thanks for the feedback and sorry for taking a while to  add it. This should be ready for another look now!

**Comment by rhatdan** [comment]
LGTM
/approve
Thanks @markusthoemmes

**Comment by rhatdan** [comment]
@jwhonce PTAL

**Comment by rhatdan** [comment]
@mheon PTAL

**Pull request merged**